### PR TITLE
PROD-608: Add valid burn sign constraint

### PR DESCRIPTION
**Pull request by neel-ds**
- Description
This PR will prevent empty string burn tx in database and record in Sentry. It adds a constraint and validate in server actions using bs58 decode if string is valid signature or not

- What are the steps to test that this code is working?
NA

- Screen shots or recordings for UI changes
NA

### CODE REVIEW

File: app/actions/chompResult.ts [hunk 1]
```diff
@@ -171,6 +175,18 @@ export async function revealQuestions(
     return null;
   }
 
+  if (!isValidSignature(burnTx)) {
```
Comment (yash431garg) [inline app/actions/chompResult.ts:178]: Can you do this before we do hasBonkBurnedCorrectly?

File: prisma/migrations/20241223180534_burn_valid_signature/migration.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE "ChompResult" 
+ADD CONSTRAINT "burnTransactionSignature_not_empty" 
+CHECK ("burnTransactionSignature" IS NULL OR LENGTH("burnTransactionSignature") > 0);
```
Comment (marvinmarnold) [inline prisma/migrations/20241223180534_burn_valid_signature/migration.sql:3]: Looking at ChompResults, burnTransactionSignature should always be between 86-88 characters.

Comment (neel-ds) [reply prisma/migrations/20241223180534_burn_valid_signature/migration.sql:3]: Do we want to limit signature length between this range?

File: app/utils/solana.ts [hunk 3]
```diff
@@ -132,3 +133,16 @@ export const getSolBalance = async (address: string): Promise<number> => {
 
   return balance / LAMPORTS_PER_SOL;
 };
+
+export function isValidSignature(
+  signature: string | null | undefined,
+): boolean {
+  if (!signature) return true; // Null is allowed but no empty string or random string
```
Comment (samott) [inline app/utils/solana.ts:140]: doesn't this allow empty strings, contrary to the comment?

Comment (marvinmarnold) [reply app/utils/solana.ts:140]: Screenshot 2024-12-23 at 11 08 24 AM

**Review by samott — DISMISSED** [review]

**Comment by neel-ds** [comment]
Gentle Reminder: We need to remove existing invalid burn tx records from prod before merging this.

File: app/utils/solana.ts [hunk 4]
```diff
@@ -132,3 +133,19 @@ export const getSolBalance = async (address: string): Promise<number> => {
 
   return balance / LAMPORTS_PER_SOL;
 };
+
+export function isValidSignature(
+  signature: string | null | undefined,
+): boolean {
+  // Null is allowed but no empty string or random string
+  if (signature === null || signature === undefined) return true;
+  if (signature.length < 85 || signature.length > 89 || signature === "")
```
Comment (marvinmarnold) [inline app/utils/solana.ts:142]: signature === "" is unnecessary now

**Pull request merged**